nixos/modules/security/doas.nix

```
@@ -236,57 +245,67 @@ in
 
   ###### implementation
 
-  config = mkIf cfg.enable {
-
-    security.doas.extraRules = mkOrder 600 [
-      {
-        groups = [ "wheel" ];
-        noPass = !cfg.wheelNeedsPassword;
-      }
-    ];
-
-    security.wrappers.doas =
-      { setuid = true;
-        owner = "root";
-        group = "root";
-        source = "${doas}/bin/doas";
-      };
-
-    environment.systemPackages = [
-      doas
-    ];
-
-    security.pam.services.doas = {
-      allowNullPassword = true;
-      sshAgentAuth = true;
-    };
-
-    environment.etc."doas.conf" = {
-      source = pkgs.runCommand "doas-conf"
+  config = mkMerge [
+    (mkIf cfg.enable {
+      security.doas.extraRules = mkOrder 600 [
         {
-          src = pkgs.writeText "doas-conf-in" ''
-            # To modify this file, set the NixOS options
-            # `security.doas.extraRules` or `security.doas.extraConfig`. To
-            # completely replace the contents of this file, use
-            # `environment.etc."doas.conf"`.
-
-            # extraRules
-            ${concatStringsSep "\n" (lists.flatten (map mkRule cfg.extraRules))}
-
-            # extraConfig
-            ${cfg.extraConfig}
-
-            # "root" is allowed to do anything.
-            permit nopass keepenv root
-          '';
-          preferLocalBuild = true;
+          groups = [ "wheel" ];
+          noPass = !cfg.wheelNeedsPassword;
         }
-        # Make sure that the doas.conf file is syntactically valid.
-        "${pkgs.buildPackages.doas}/bin/doas -C $src && cp $src $out";
-      mode = "0440";
-    };
+      ];
+
+      security.wrappers.doas =
+        { setuid = true;
+          owner = "root";
+          group = "root";
+          source = "${cfg.package}/bin/doas";
+        };
+
+      environment.systemPackages = [
+        cfg.package
+      ];
+
+      security.pam.services.doas = {
+        allowNullPassword = true;
+        sshAgentAuth = true;
+      };
 
-  };
+      environment.etc."doas.conf" = {
+        source = pkgs.runCommand "doas-conf"
+          {
+            src = pkgs.writeText "doas-conf-in" ''
+              # To modify this file, set the NixOS options
+              # `security.doas.extraRules` or `security.doas.extraConfig`. To
+              # completely replace the contents of this file, use
+              # `environment.etc."doas.conf"`.
```

```suggestion
              # To modify this file, set the NixOS options `security.doas.extraRules` or `security.doas.extraConfig`.
              # To completely replace the contents of this file, use `environment.etc."doas.conf"`.
```
lets make that a bit more readable, we don't need to be afraid of slightly over 80 characters